x-pack/plugins/security_solution/public/timelines/components/timeline/header/title_and_description.tsx

```diff
@@ -109,34 +115,65 @@ export const TimelineTitleAndDescription = React.memo<TimelineTitleAndDescriptio
       return i18n.CLOSE_MODAL;
     }, [showWarning, status, timelineType]);
 
-    useEffect(() => {
-      if (isFormSubmitted && !isSaving && prevIsSaving) {
-        closeSaveTimeline();
-      }
-    }, [isFormSubmitted, isSaving, prevIsSaving, closeSaveTimeline]);
-
-    const modalHeader =
-      status === TimelineStatus.draft
-        ? timelineType === TimelineType.template
-          ? i18n.SAVE_TIMELINE_TEMPLATE
-          : i18n.SAVE_TIMELINE
-        : timelineType === TimelineType.template
-        ? i18n.NAME_TIMELINE_TEMPLATE
-        : i18n.NAME_TIMELINE;
-
-    const saveButtonTitle =
-      status === TimelineStatus.draft && showWarning
-        ? timelineType === TimelineType.template
-          ? i18n.SAVE_TIMELINE_TEMPLATE
-          : i18n.SAVE_TIMELINE
-        : i18n.SAVE;
+    const modalHeader = useMemo(
+      () =>
+        status === TimelineStatus.draft
+          ? timelineType === TimelineType.template
+            ? i18n.SAVE_TIMELINE_TEMPLATE
+            : i18n.SAVE_TIMELINE
+          : timelineType === TimelineType.template
+          ? i18n.NAME_TIMELINE_TEMPLATE
+          : i18n.NAME_TIMELINE,
+      [status, timelineType]
+    );
+
+    const saveButtonTitle = useMemo(
+      () =>
+        status === TimelineStatus.draft && showWarning
+          ? timelineType === TimelineType.template
+            ? i18n.SAVE_TIMELINE_TEMPLATE
+            : i18n.SAVE_TIMELINE
+          : i18n.SAVE,
+      [showWarning, status, timelineType]
+    );
 
     const calloutMessage = useMemo(() => i18n.UNSAVED_TIMELINE_WARNING(timelineType), [
       timelineType,
     ]);
 
-    const descriptionLabel =
-      status === TimelineStatus.draft ? `${DESCRIPTION} (${OPTIONAL})` : DESCRIPTION;
+    const descriptionLabel = useMemo(() => `${i18n.TIMELINE_DESCRIPTION} (${i18n.OPTIONAL})`, []);
+
+    const titleFieldProps = useMemo(
+      () => ({
+        'aria-label': i18n.TIMELINE_TITLE,
+        autoFocus: initialFocus === 'title',
+        'data-test-subj': 'save-timeline-title',
+        disabled: isSaving,
+        spellCheck: true,
+        placeholder:
+          timelineType === TimelineType.template
+            ? commonI18n.UNTITLED_TEMPLATE
+            : commonI18n.UNTITLED_TIMELINE,
+      }),
+      [initialFocus, isSaving, timelineType]
+    );
+
+    const descriptionFieldProps = useMemo(
+      () => ({
+        'aria-label': i18n.TIMELINE_DESCRIPTION,
+        autoFocus: initialFocus === 'description',
+        'data-test-subj': 'save-timeline-description',
+        disabled: isSaving,
+        placeholder: commonI18n.DESCRIPTION,
+      }),
+      [initialFocus, isSaving]
+    );
+
+    useEffect(() => {
+      if (isSubmitted && !isSaving && prevIsSaving) {
+        closeSaveTimeline();
```

Comment:
nit -> with this new way, what do you think about closing the modal in the `handleSubmit`? I Think it will be better now!

Reply:
not really, because our actions are not async, so we will be closing modal before the timeline is saved

Reply:
ok, I thought that we wanted to have a progressive UX here. I understand what you are saying, we can re-visit it later on that's not a blocker to me